### PR TITLE
Add Vector.ScaleVec

### PR DESCRIPTION
**Pull request by vladimir-ch**
Is it ok like this?

### CODE REVIEW

**Comment by kortschak** [comment]
The signature doesn't match the convention, it should be v.Scale(alpha float64, a *Vector) where the dot is read as an assignment.

The full work would be to check a.Len() == v.Len() or v.isZero(), or a == v, then as appropriate copy a to v (not if a == v) and then call Scal on v as you have here.

**Comment by vladimir-ch** [comment]
Changed as suggested. PTAL @kortschak

**Comment by kortschak** [comment]
LGTM

File: mat64/vector_test.go [hunk 1]
```diff
@@ -134,6 +134,66 @@ func TestVectorMul(t *testing.T) {
...
+		if !reflect.DeepEqual(v.RawVector(), test.want.RawVector()) {
+			t.Errorf("test %d: unexpected result for v = alpha * a: got: %v want: %v", i, v.RawVector(), test.want.RawVector())
+		}
+
+		v.CopyVec(test.a)
+		v.ScaleVec(test.alpha, &v)
+		if !reflect.DeepEqual(v.RawVector(), test.want.RawVector()) {
+			t.Errorf("test %d: unexpected result for v = alpha * v: got: %v want: %v", i, v.RawVector(), test.want.RawVector())
+		}
+	}
```
Comment (btracey) [inline mat64/vector_test.go]: Could you add a call to testOneInput as well for testing please?

Comment (vladimir-ch) [inline mat64/vector_test.go]: Using `testOneInput` for the first time, so could you please take another look, @btracey ?

File: mat64/vector_test.go [hunk 2]
```diff
@@ -134,6 +134,77 @@ func TestVectorMul(t *testing.T) {
...
+
+	for _, alpha := range []float64{0, 1, -1, 2.3, -2.3} {
+		method := func(receiver, a Matrix) {
+			v := receiver.(*Vector)
+			v.ScaleVec(alpha, a.(*Vector))
```
Comment (btracey) [inline mat64/vector_test.go]: We've been using interfaces here, see for example TestVectorMul

File: mat64/vector_test.go [hunk 3]
```diff
@@ -134,6 +134,77 @@ func TestVectorMul(t *testing.T) {
...
+		}
+		denseComparison := func(receiver, a *Dense) {
+			receiver.Scale(alpha, a)
+		}
+		testOneInput(t, "ScaleVec", &Vector{}, method, denseComparison, legalTypeVec, isAnySize, 0)
```
Comment (btracey) [inline mat64/vector_test.go]: This works, but sort of by accident. I'd rather see a "isAnyVector" than isAnySize, that way the fact that it must be a vector is explicitly declared in the size(

Comment (vladimir-ch) [inline mat64/vector_test.go]: Ok, I'll add it. At first I didn't because `legalDims` does that and it felt like code duplication.

Comment (btracey) [inline mat64/vector_test.go]: Yea, I don't know. On the one hand, you're definitely right. On the other hand, isAnySize is generally for matrices, so it's somewhat confusing when reading other versions of the test. Not clear which is better.

Comment (kortschak) [inline mat64/vector_test.go]: Being overly tight I think is the way to go - see the obsessive name changes and nilling I did recently.

Comment (vladimir-ch) [inline mat64/vector_test.go]: Done. It is probably better to be explicit. PTAL @btracey

File: mat64/list_test.go [hunk 4]
```diff
@@ -66,6 +66,11 @@ func isAnySize2(ar, ac, br, bc int) bool {
 	return true
 }
 
+// isAnyVector returns true for any row or column vector sizes.
```
Comment (btracey) [inline mat64/list_test.go]: I think this should just check if the number of columns is 1. *Vector is always a column.

**Comment by btracey** [comment]
LGTM with stricter definition of "AnyVector"

File: mat64/list_test.go (hunk 4 above)
Comment (vladimir-ch) [inline mat64/list_test.go]: Done

**Pull request merged**